### PR TITLE
Make KNN feature normalization optional

### PR DESCRIPTION
**Pull request by guarin**
## Changes

* Make KNN feature normalization optional
* Always reset KNN features at beginning of training epoch

## How was it tested?
* Added unit tests

Closes: #1425

### CODE REVIEW

File: lightly/utils/benchmarking/knn_classifier.py [hunk 1]
```diff
@@ -145,6 +153,12 @@ def on_train_epoch_start(self) -> None:
         # Set model to eval mode to disable norm layer updates.
         self.model.eval()
 
+        # Reset features and targets.
+        self._train_features = []
+        self._train_targets = []
+        self._train_features_tensor = None
+        self._train_targets_tensor = None
```
Comment (guarin) [inline lightly/utils/benchmarking/knn_classifier.py:160]: This is not strictly necessary because one should not "train" the KNNClassifier for more than one epoch. But I still added it to avoid any issues if someone actually uses it over multiple epochs.

**Review by philippmwirth — APPROVED** [review]
